vspace/src/main/java/edu/asu/diging/vspace/web/staff/AddTextBlockController.java

```diff
@@ -19,14 +21,15 @@
     @Autowired
     private IContentBlockManager contentBlockManager;
 
-    @RequestMapping(value = "/staff/module/{moduleId}/slide/{id}/textcontent", method = RequestMethod.POST)
-    public ResponseEntity<String> addTextBlock(@PathVariable("id") String slideId,
+    @RequestMapping(value = "/staff/module/{moduleId}/slide/{id}/textcontent", method = RequestMethod.POST, produces = "application/json")
+    public ResponseEntity<HashMap> addTextBlock(@PathVariable("id") String slideId,
             @PathVariable("moduleId") String moduleId, @RequestParam("content") String content,
             @RequestParam("contentOrder") Integer contentOrder) throws IOException {
 
-        contentBlockManager.createTextBlock(slideId, content, contentOrder);
-
-        return new ResponseEntity<String>(HttpStatus.OK);
+        ITextBlock textBlock = contentBlockManager.createTextBlock(slideId, content, contentOrder);
+        HashMap<String, String> data = new HashMap<String, String>();
```

Comment:
responseEntity should be of type 'Map' (the interface) not hashmap. And the declaration here should also use 'Map'.